ddtrace/contrib/vertica/patch.py

```diff
@@ -0,0 +1,265 @@
+                    #  }
+                    # and an instance `c`:
+                    # c = C()
+                    # c.my_method()
+                    #
+                    # An instance of either will have a pin attached so that
+                    # the user can override specific configuration for that
+                    # instance.
+                    #
+                    # The pins will contain the config for the instance which
+                    # they are attached:
+                    #   PinB._config == { 'routines: { 'my_method: { ... } } }
+                    #   PinC._config == { }
+                    #
+                    # The problem here is that at this point in the wrapper
+                    # when c.my_method() is called we have an instance of C and
+                    # a pin from C, but require the config from B.
+                    #
+                    # Possible solution: follow inheritance and look for config
+                    # on a parent. Can we use normal python inheritance somehow?
+                    if _patch_routine in pin._config["routines"]:
+                        conf = pin._config["routines"][_patch_routine]
+                    else:
+                        conf = _find_routine_config(config, instance, _patch_routine)
+
+                    enabled = conf.get("trace_enabled", True)
+
+                    span = None
+
+                    try:
+                        # shortcut if not enabled
+                        if not enabled:
+                            result = wrapped(*args, **kwargs)
+                            return result
+
+                        operation_name = conf["operation_name"]
+                        tracer = pin.tracer
+                        with tracer.trace(operation_name, service=pin.service) as span:
+                            span.set_tags(pin.tags)
+                            if "span_type" in conf:
+                                span.span_type = conf["span_type"]
+
+                            if "span_start" in conf:
+                                conf["span_start"](
+                                    instance, span, conf, *args, **kwargs
+                                )
+
+                            result = wrapped(*args, **kwargs)
+                            return result
+                    except Exception as err:
+                        if "on_error" in conf:
+                            conf["on_error"](instance, err, span, conf, *args, **kwargs)
+                        raise
+                    finally:
+                        # if an exception is raised result will not exist
+                        if "result" not in locals():
+                            result = None
+                        if "span_end" in conf:
+                            conf["span_end"](
+                                instance, result, span, conf, *args, **kwargs
+                            )
+
+            wrap_routine()
```

Comment:
Same here, why are we making this a function rather than calling the code immediately?

Reply:
I assume the goal is probably to close over some local variables? In which case we should take those variables in as parameters.

```python
def wrap_routine(patch_mod, patch_class, patch_routine, ...):
    pass
```

Doing this will make the code more clear to people reading it and will help ensure that we truly are closing over those variables.